### PR TITLE
Use Databricks SDK for OAuth

### PR DESCRIPTION
**Pull request by B-Step62**
In the test workspace, we need to use service principle based authentication using OAuth, not PAT. Replacing the raw HTTP request to Databricks SDK to handle it easily (yes I ended up using it🙃).

### CODE REVIEW

File: libs/databricks/tests/integration_tests/test_vectorstore.py [hunk 1]
```diff
@@ -23,62 +24,25 @@ def test_vectorstore():
...
-    # class is also migrated to the SDK.
-    response = requests.post(
-        f"{test_endpoint}/api/2.1/jobs/run-now",
-        json={
-            "job_id": test_job_id,
-        },
-        headers=headers,
+    response = w.jobs.run_now(job_id=test_job_id)
+    job_url = (
+        f"{os.getenv('DATABRICKS_HOST')}/jobs/{test_job_id}/runs/{response.run_id}"
```
Comment (harupy) [inline libs/databricks/tests/integration_tests/test_vectorstore.py:41]: Can we use `client.config.host` to get the host?

```python
>>> w = WorkspaceClient()
>>> w
WorkspaceClient(host='https://foo.cloud.databricks.com', auth_type='pat', ...)
>>> w.config
<Config: host=https://foo.cloud.databricks.com, token=***, auth_type=pat>
>>> w.config.host
'foo.cloud.databricks.com'
```

File: libs/databricks/tests/integration_tests/test_vectorstore.py [hunk 2]
```diff
@@ -23,62 +24,25 @@ def test_vectorstore():
...
-    # TODO: We are going to replace this with the Databricks SDK once the vector store
-    # class is also migrated to the SDK.
-    response = requests.post(
-        f"{test_endpoint}/api/2.1/jobs/run-now",
-        json={
-            "job_id": test_job_id,
-        },
-        headers=headers,
+    response = w.jobs.run_now(job_id=test_job_id)
+    job_url = (
```
Comment (harupy) [inline libs/databricks/tests/integration_tests/test_vectorstore.py:40]: Can we use https://databricks-sdk-py.readthedocs.io/en/latest/dbdataclasses/jobs.html#databricks.sdk.service.jobs.Run.run_page_url?

File: libs/databricks/tests/integration_tests/test_vectorstore.py [hunk 3]
```diff
@@ -23,62 +24,25 @@ def test_vectorstore():
...
-        status = response.json()["status"]
-        if status["state"] == "TERMINATED":
-            if status["termination_details"]["type"] == "SUCCESS":
-                break
-            else:
-                assert False, "Job failed. Please check the logs in the workspace."
-
-        time.sleep(60)
-        print("Job is still running...")  # noqa: T201
+    result = response.result(timeout=timedelta(seconds=3600))
```
Comment (harupy) [inline libs/databricks/tests/integration_tests/test_vectorstore.py:46]: can we use https://databricks-sdk-py.readthedocs.io/en/latest/workspace/jobs/jobs.html#databricks.sdk.service.jobs.JobsAPI.run_now_and_wait?

File: libs/databricks/tests/integration_tests/test_vectorstore.py (hunk 1 above)
Comment (B-Step62) [reply libs/databricks/tests/integration_tests/test_vectorstore.py:41]: oh nice, didn't know we can do that!

File: libs/databricks/tests/integration_tests/test_vectorstore.py (hunk 2 above)
Comment (B-Step62) [reply libs/databricks/tests/integration_tests/test_vectorstore.py:40]: nice utility!

File: libs/databricks/tests/integration_tests/test_vectorstore.py (hunk 3 above)
Comment (B-Step62) [reply libs/databricks/tests/integration_tests/test_vectorstore.py:46]: Splitting to show run URL before completion. The [run_now_and_wait](https://github.com/databricks/databricks-sdk-py/blob/e23b4acd0cc967ee2a3cba3c3d26666698600e35/databricks/sdk/service/jobs.py#L6329) API internally just to `run_now(...).result()`.

Comment (harupy) [reply libs/databricks/tests/integration_tests/test_vectorstore.py:46]: got it :)

File: libs/databricks/tests/integration_tests/test_vectorstore.py (hunk 2 above)
Comment (B-Step62) [reply libs/databricks/tests/integration_tests/test_vectorstore.py:40]: Hmm wait we only get `Wait` object from the `run_now` API and it only provide access to the `run_id` field.

https://github.com/databricks/databricks-sdk-py/blob/4e751f88a605a7deacc8f411240dd6631f21936f/databricks/sdk/service/jobs.py#L5833C16-L5835

**Review by harupy — APPROVED** [review]
LGTM!

**Pull request merged**